src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -43,6 +46,8 @@ internal class EvalVisitor : IRNodeVisitor<ValueTask<FormulaValue>, EvalVisitorC
 
         public Governor Governor { get; private set; }
 
+        private readonly Stack<FormulaValue[]> _formulaValuesStack = new Stack<FormulaValue[]>();

```

Comment:
>_formulaValuesStack [](http://example.com/codeflow?start=47&length=19)

this is good. 
Let's take it a step further and wrap in something like UdfStackFrame (see chat)